rpc/src/lib.rs

```diff
@@ -226,13 +231,25 @@ pub fn start_ipc<M, S, H, T>(
 	addr: &str,
 	handler: H,
 	extractor: T,
+	chmod: u16
 ) -> ::std::io::Result<ipc::Server> where
 	M: jsonrpc_core::Metadata,
 	S: jsonrpc_core::Middleware<M>,
 	H: Into<jsonrpc_core::MetaIoHandler<M, S>>,
 	T: IpcMetaExtractor<M>,
 {
+	#[cfg(target_os = "macos")]
+	let chmod = u16::from_str_radix(&format!("{}", chmod), 8)
+		.expect("chmod is a u16; qed");
+	#[cfg(not(target_os = "macos"))]
+	let chmod = u32::from_str_radix(&format!("{}", chmod), 8)
+		.expect("chmod is a u16; qed");
```

Comment:
This stuff merits a comment, what is happening here? Does `set_mode()` take a `u16` on bsds and [`u32` on linux](https://docs.rs/jsonrpc-ipc-server/14.0.5/jsonrpc_ipc_server/struct.SecurityAttributes.html#method.set_mode)?

Reply:
yeah basically.

Reply:
can't we do (casting u16 to u32 is safe):

```rustc
   let chmod = u16::from_str_radix(&format!("{}", chmod), 8);
   let attr = SecurityAttributes::empty().set_mode(chmod as _)?;
```
to avoid the extra conditional compilation stuff?